src/model/user.ts

```diff
@@ -0,0 +1,66 @@
+import mongoose, { Schema } from "mongoose"
+
+const rawSchema = {
+    _id: {
+        type: String,
+        required: true,
+        validate: {
+            validator: function (input: string) {
+                return /^U\d+$/g.test(input)
+            },
+
+            message: "User IDs must have the format /^U\\d+$/g",
+        },
+    },
+
+    name: {
+        type: String,
+        required: true,
+    },
+
+    profileImage: {
+        type: String,
+        required: true,
```

Comment:
Could probably specify an empty string to signifying default profile image

Reply:
Then it should be setup by default on server or can just use mongoose [default](https://mongoosejs.com/docs/defaults.html) to remove errors when `profileImage` are not provided.

Reply:
dibuat optional aja buat profileImage

Reply:
Is it not better to have it explicit from FE saying "this guy did not upload a pic, so use this empty string as the pic" instead of BE thinking "no pic was sent, idk if they forgot to or it got lost or whatever so assume default pic"

Reply:
I think it's better to handle and minimize error returned _In Case_ there are other type of client in the future (Eg: bgdc bot that integrate with the site backend that will add user or something) or lost as you mention. though it is good to validate both FE and BE.

well i guess it still ok to label it as required. Resolving this conv

Reply:
Fixed on #4